addOns/grpc/src/main/java/org/zaproxy/addon/grpc/internal/VariantGrpc.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.grpc.internal;
+
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.core.scanner.NameValuePair;
+import org.parosproxy.paros.core.scanner.Variant;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class VariantGrpc implements Variant {
+    private static final Logger LOGGER = LogManager.getLogger(VariantGrpc.class);
+    private final List<NameValuePair> params = new ArrayList<>();
+    private final ProtoBufMessageEncoder protoBufMessageEncoder = new ProtoBufMessageEncoder();
+    private final ProtoBufMessageDecoder protoBufMessageDecoder = new ProtoBufMessageDecoder();
+
+    @Override
+    public void setMessage(HttpMessage msg) {
+        String contentType = msg.getRequestHeader().getHeader("content-type");
+        if (contentType != null && isValidContentType(contentType)) {
+            try {
+                byte[] body = Base64.getDecoder().decode(msg.getRequestBody().getBytes());
+                byte[] payload = DecoderUtils.extractPayload(body);
+                this.protoBufMessageDecoder.decode(payload);
+                this.parseContent(protoBufMessageDecoder.getDecodedToList());
+            } catch (Exception e) {
+                LOGGER.error("Parsing message body failed: {}", e.getMessage());
+            }
+        }
+    }
+
+    public void parseContent(List<String> decodedList) {
+        for (String pair : decodedList) {
+            String[] nameValuePair = pair.split("::", 2);
+            params.add(
+                    new NameValuePair(
+                            NameValuePair.TYPE_JSON,
+                            nameValuePair[0],
+                            nameValuePair[1],
+                            params.size()));
+        }
+    }
+
+    public boolean isValidContentType(String contentType) {
+        return contentType.startsWith("application/grpc-web-text");
+    }
+
+    @Override
+    public List<NameValuePair> getParamList() {
+        return params;
+    }
+
+    @Override
+    public String setParameter(
+            HttpMessage msg, NameValuePair originalPair, String param, String value) {
+        NameValuePair currentPair = params.get(originalPair.getPosition());
+        String reqBody = buildNewBodyContent(currentPair, param, value);
+        try {
+            setEncodedReqBodyMessage(msg, reqBody);
+            return value;
+        } catch (Exception e) {
+            LOGGER.warn("Failed to set parameter in GraphQL message: {}", e.getMessage());
```

Comment:
😉

Reply:
This is about the log message that still mentions the other format.